src/planscape/planning/tests/test_scenario_views.py

```diff
@@ -0,0 +1,1175 @@
+                "configuration": self.configuration,
+                "name": "test scenario",
+                "notes": "test notes",
+            }
+        )
+        response = self.client.post(
+            reverse("planning:create_scenario"),
+            payload,
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        output = response.json()
+        scenario_id = output["id"]
+        self.assertEqual(Scenario.objects.count(), 1)
+        self.assertEqual(ScenarioResult.objects.count(), 1)
+        scenario = Scenario.objects.get(pk=scenario_id)
+        self.assertEqual(scenario.planning_area.pk, self.planning_area.pk)
+        self.assertEqual(scenario.configuration, self.configuration)
+        self.assertEqual(scenario.name, "test scenario")
+        self.assertEqual(scenario.notes, "test notes")
+        self.assertEqual(scenario.user, self.user)
+
+    @mock.patch(
+        "planning.views.validate_scenario_treatment_ratio",
+        return_value=(True, "all good"),
+    )
+    def test_create_scenario_no_notes(self, validation):
+        self.client.force_authenticate(self.user)
+        payload = json.dumps(
+            {
+                "planning_area": self.planning_area.pk,
+                "configuration": self.configuration,
+                "name": "test scenario",
+            }
+        )
+        response = self.client.post(
+            reverse("planning:create_scenario"),
+            payload,
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 200)
+        output = json.loads(response.content)
+        scenario_id = output["id"]
+        self.assertEqual(Scenario.objects.count(), 1)
+        self.assertEqual(ScenarioResult.objects.count(), 1)
+        scenario = Scenario.objects.get(pk=scenario_id)
+        self.assertEqual(scenario.planning_area.pk, self.planning_area.pk)
+        self.assertEqual(scenario.configuration, self.configuration)
+        self.assertEqual(scenario.name, "test scenario")
+        self.assertEqual(scenario.notes, None)
+        self.assertEqual(scenario.user, self.user)
+
+    def test_create_scenario_missing_planning_area(self):
+        self.client.force_authenticate(self.user)
+        payload = json.dumps(
+            {"configuration": self.configuration, "name": "test scenario"}
+        )
+        response = self.client.post(
+            reverse("planning:create_scenario"),
+            payload,
+            content_type="application/json",
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertRegex(str(response.content), r"This field is required")
+
+    def test_create_scenario_missing_configuration(self):
+        self.client.force_authenticate(self.user)
+        payload = json.dumps(
+            {"planning_area": self.planning_area.pk, "name": "test scenario"}
+        )
+        response = self.client.post(
```

Comment:
similar to my other comment, this block is repeated a bunch of times, maybe its worth a helper function

Reply:
I was thinking that, too. I'll see if there's a clean way to split that up. Thanks for the thoughts and comments!